### PR TITLE
Chongyangbai/multilabel2kvp

### PR DESCRIPTION
**Pull request by cy-bai**
multilabel ic to kvp adaptor

### CODE REVIEW

File: tests/test_fixtures.py [hunk 1]
```diff
@@ -98,3 +98,44 @@ def create_an_ic_manifest(root_dir='', n_images=2, n_categories=3):
...
+            "files_for_local_usage": [
+                "train.zip"
+            ]
+        },
+    }
+
+    @staticmethod
+    def create_an_ic_dataset(n_images=2, n_categories=3):
+        dataset_dict = copy.deepcopy(MultilabelClassificationTestFixtures.DATASET_INFO_DICT)
+        tempdir = tempfile.TemporaryDirectory()
```
Comment (shonohs) [inline tests/test_fixtures.py:118]: This TemporaryDirectory() is not removed. Maybe change this method to context manager with `@contextlib.contextmanager`?

File: vision_datasets/image_classification/classification_as_kvp_dataset.py [hunk 2]
```diff
@@ -70,20 +55,79 @@ def __init__(self, classification_dataset: VisionDataset):
...
-        Convert the classification dataset label_name to the desired format for KVP annnotation as defined by the BASE_CLASSIFICATION_SCHEMA.
+        Convert the classification dataset label_name to the desired format for KVP annnotation as defined by the SCHEMA_BASE.
         E.g. {"fields": {"className": {"value": <label_name>}}}
 
         """
         return {
             f"{KeyValuePairLabelManifest.LABEL_KEY}": {
-                f"{CLASS_NAME_KEY}": {
-                    f"{KeyValuePairLabelManifest.LABEL_VALUE_KEY}": label_name
+                f"{MultiClassAsKeyValuePairDataset.CLASS_NAME_KEY}": {
```
Comment (shonohs) [inline vision_datasets/image_classification/classification_as_kvp_dataset.py]: I see this `f"{variable}"` pattern in many places. Why don't we simply use the variable?

**Review by nisyad-ms — APPROVED** [review]

**Comment by nisyad-ms** [comment]
Looks good to me.

File: vision_datasets/image_classification/classification_as_kvp_dataset.py [hunk 3]
```diff
@@ -70,20 +55,79 @@ def __init__(self, classification_dataset: VisionDataset):
...
     def construct_schema(self, class_names: typing.List[str]) -> typing.Dict[str, typing.Any]:
-        schema: typing.Dict[str, typing.Any] = BASE_CLASSIFICATION_SCHEMA  # initialize with base schema
-        schema["fieldSchema"][f"{CLASS_NAME_KEY}"]["classes"] = {c: {"description": f"A single class name. Only output {c} as the class name if present."} for c in class_names}
+        raise NotImplementedError
+
+    def construct_kvp_label_data(self, label_names: typing.List[str]) -> typing.Dict[str, typing.Union[typing.Dict[str, typing.Dict[str, str]], None]]:
+        raise NotImplementedError
+
+
+class MultiClassAsKeyValuePairDataset(ClassificationAsKeyValuePairDatasetBase):
```
Comment (juliagongms) [inline vision_datasets/image_classification/classification_as_kvp_dataset.py]: Nit: maybe `MulticlassClassificationAsKeyValuePairDataset`? Same for `MultilabelClassificationAsKeyValuePairDataset` (for clarity, outside of the context of `ClassificationAsKeyValuePairDatasetBase`).

**Review by juliagongms — APPROVED** [review]
Added a small comment. Looks good otherwise, thanks!

File: tests/test_fixtures.py (hunk 1 above)
Comment (cy-bai) [reply tests/test_fixtures.py:118]: thanks, added `with tempdir` statement when calling this function so the tempdir will be automatically deleted

**Review by juliagongms — APPROVED** [review]

**Review by shonohs — APPROVED** [review]

**Pull request merged**